fml/message_loop_task_queues.cc

```diff
@@ -336,32 +358,28 @@ TaskSource::TopTask MessageLoopTaskQueues::PeekNextTaskUnlocked(
     TaskQueueId owner) const {
   FML_DCHECK(HasPendingTasksUnlocked(owner));
   const auto& entry = queue_entries_.at(owner);
-  const TaskQueueId subsumed = entry->owner_of;
-  if (subsumed == _kUnmerged) {
+  if (entry->owner_of.empty()) {
     return entry->task_source->Top();
   }
 
   TaskSource* owner_tasks = entry->task_source.get();
-  TaskSource* subsumed_tasks = queue_entries_.at(subsumed)->task_source.get();
-
-  // we are owning another task queue
-  const bool subsumed_has_task = !subsumed_tasks->IsEmpty();
-  const bool owner_has_task = !owner_tasks->IsEmpty();
-  fml::TaskQueueId top_queue_id = owner;
-  if (owner_has_task && subsumed_has_task) {
-    const auto owner_task = owner_tasks->Top();
-    const auto subsumed_task = subsumed_tasks->Top();
-    if (owner_task.task > subsumed_task.task) {
-      top_queue_id = subsumed;
-    } else {
-      top_queue_id = owner;
+  std::vector<TaskSource::TopTask> candidate_top_tasks;
+
+  if (!owner_tasks->IsEmpty()) {
+    candidate_top_tasks.push_back(owner_tasks->Top());
+  }
+  for (TaskQueueId subsumed : entry->owner_of) {
+    TaskSource* subsumed_tasks = queue_entries_.at(subsumed)->task_source.get();
+    if (!subsumed_tasks->IsEmpty()) {
+      candidate_top_tasks.push_back(subsumed_tasks->Top());
     }
-  } else if (owner_has_task) {
-    top_queue_id = owner;
-  } else {
-    top_queue_id = subsumed;
   }
-  return queue_entries_.at(top_queue_id)->task_source->Top();
+  // At least one task at the top because PeekNextTaskUnlocked() is called after
+  // HasPendingTasksUnlocked()
+  FML_CHECK(!candidate_top_tasks.empty());
+  TaskSource::TopTask& top =
+      *std::min_element(candidate_top_tasks.begin(), candidate_top_tasks.end());
```

Comment:
Instead of relying on operator overloading here, just pass in a compare function.